SalesforceCommerceCloud/pwa-kit
# @W-16675302@ : Remove ocapi session-bridging on phased launches

# Description

Based on a recent POC for appserver affinity, we were able to prove that after a recent change in Plugin SLAS where we now send the access_token from SFRA to PWA Kit via cc_at cookie. With this, we no longer need to make calls to OCAPI `/sessions` from PWA Kit and can improve stability of hybrid storefronts by reducing session churn.

As a Hybrid storefront stakeowner, I'd like to remove unnecessary calls to OCAPI session-bridge from PWA Kit.

**NOTE**: This change has corresponding changes in Plugin SLAS. See: https://github.com/SalesforceCommerceCloud/plugin_slas/pull/210



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Remove prop OCAPISessionsURL from CommerceApiProvider.
- Remove implementation and function calls to createOCAPISession() in commerce-sdk-react.
- Update commerce-sdk-react/auth to clear dwsid cookie whenever a shopper logs (guest to registered) in or logs out(registered to guest).

# How to Test-Drive This PR

- Navigate to www.phased-launch-testing.com
- Open the Network Tab in Dev Tools and confirm no requests are being made to OCAPI /sessions endpoint
- Run hybrid test suite.

# Hybrid Test Suite Results [Run Locally]
<img width="1008" alt="Screenshot 2024-09-17 at 10 45 26 AM" src="https://github.com/user-attachments/assets/90e1d1fe-19fd-4a43-9594-0ef15a241a18">

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
